### ISSUE TITLE
Unable to pull docker images of  microsoft/nanoserver or 

### ISSUE THREAD

**Issue by JohanSpannare** [docker/hub-feedback#874]
Anyone else having the same problem?

Its only retrying over and over again...

Same for microsoft/windowsservercore

**Comment by JohanSpannare** [comment]
`time="2016-11-23T14:14:06.664996649Z" level=debug msg="Calling GET /v1.24/info"
time="2016-11-23T14:14:06.667096922Z" level=debug msg="Calling POST /v1.24/images/create?fromImage=microsoft%2Fnanoserver&tag=latest"
time="2016-11-23T14:14:06.667232360Z" level=debug msg="Trying to pull microsoft/nanoserver from https://registry-1.docker.io v2"
time="2016-11-23T14:14:07.901491865Z" level=debug msg="Increasing token expiration to: 60 seconds"
time="2016-11-23T14:14:08.517656820Z" level=debug msg="Pulling ref from V2 registry: microsoft/nanoserver:latest"
time="2016-11-23T14:14:08.518671295Z" level=debug msg="pulling blob \"sha256:5496abde368a3dd39999745bf998c877ddc6a390a943bc3fd99ffaabf728ed88\""
time="2016-11-23T14:14:08.518983784Z" level=debug msg="pulling blob \"sha256:482ab31872a23b32cbdeca13edb7a0b97290714c0b5edcce96fbb3e34221ea91\""
time="2016-11-23T14:14:09.721724730Z" level=error msg="Download failed, retrying: unknown blob"
time="2016-11-23T14:14:09.887262556Z" level=error msg="Download failed, retrying: unknown blob"
time="2016-11-23T14:14:14.721843787Z" level=debug msg="pulling blob \"sha256:482ab31872a23b32cbdeca13edb7a0b97290714c0b5edcce96fbb3e34221ea91\""
time="2016-11-23T14:14:14.887446800Z" level=debug msg="pulling blob \"sha256:5496abde368a3dd39999745bf998c877ddc6a390a943bc3fd99ffaabf728ed88\""
time="2016-11-23T14:14:15.958202210Z" level=error msg="Download failed, retrying: unknown blob"
time="2016-11-23T14:14:16.215933989Z" level=error msg="Download failed, retrying: unknown blob"
time="2016-11-23T14:14:25.958411656Z" level=debug msg="pulling blob \"sha256:482ab31872a23b32cbdeca13edb7a0b97290714c0b5edcce96fbb3e34221ea91\""
time="2016-11-23T14:14:26.216728230Z" level=debug msg="pulling blob \"sha256:5496abde368a3dd39999745bf998c877ddc6a390a943bc3fd99ffaabf728ed88\""
time="2016-11-23T14:14:27.171405699Z" level=error msg="Download failed, retrying: unknown blob"
time="2016-11-23T14:14:27.477461080Z" level=error msg="Download failed, retrying: unknown blob"
time="2016-11-23T14:14:42.172569538Z" level=debug msg="pulling blob \"sha256:482ab31872a23b32cbdeca13edb7a0b97290714c0b5edcce96fbb3e34221ea91\""
time="2016-11-23T14:14:42.477774712Z" level=debug msg="pulling blob \"sha256:5496abde368a3dd39999745bf998c877ddc6a390a943bc3fd99ffaabf728ed88\""
time="2016-11-23T14:14:43.407756974Z" level=error msg="Download failed, retrying: unknown blob"
time="2016-11-23T14:14:43.737851626Z" level=error msg="Download failed, retrying: unknown blob"
time="2016-11-23T14:15:03.408309702Z" level=debug msg="pulling blob \"sha256:482ab31872a23b32cbdeca13edb7a0b97290714c0b5edcce96fbb3e34221ea91\""
time="2016-11-23T14:15:03.739059593Z" level=debug msg="pulling blob \"sha256:5496abde368a3dd39999745bf998c877ddc6a390a943bc3fd99ffaabf728ed88\""
time="2016-11-23T14:15:04.649295451Z" level=error msg="Download failed: unknown blob"
time="2016-11-23T14:15:04.649569340Z" level=error msg="Error trying v2 registry: unknown blob"
time="2016-11-23T14:15:04.649582870Z" level=error msg="Attempting next endpoint for pull after error: unknown blob"
time="2016-11-23T14:15:04.649591556Z" level=debug msg="Skipping v1 endpoint https://index.docker.io because v2 registry was detected"`

**Comment by darrenstahlmsft** [comment]
See https://github.com/docker/docker/issues/28892

**Comment by sobek85** [comment]
Seeing the same issue

**Comment by johnsimons** [comment]
Here is what I get:
```
> docker pull microsoft/nanoserver
Using default tag: latest
latest: Pulling from microsoft/nanoserver
5496abde368a: Pulling fs layer
482ab31872a2: Pulling fs layer
482ab31872a2: Verifying Checksum
482ab31872a2: Download complete
5496abde368a: Verifying Checksum
5496abde368a: Download complete
docker : failed to register layer: re-exec error: exit status 1: output: ProcessBaseLayer
C:\ProgramData\docker\windowsfilter\5f12692222e871fc80f36cf7892b4df0444d78cd211b6c5efd7df19931a131a9: This operation
returned because the timeout period expired.
    + CategoryInfo          : NotSpecified: (failed to regis...period expired.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
```

docker version:
```
Client:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64
```

**Comment by darrenstahlmsft** [comment]
@johnsimons See https://github.com/docker/docker/issues/27527

**Comment by johnsimons** [comment]
@darrenstahlmsft I have the latest windows updates.
BTW, I am running this on a Nano Server image

**Comment by darrenstahlmsft** [comment]
@johnsimons Can you open a new issue on https://github.com/docker/docker/issues with all the requested info there please? This seems like it might be a different issue than I suspected above.

**Comment by thorstenhillmann** [comment]
Found any solution? Still got the same issue on Windows Nano Server

**Comment by darrenstahlmsft** [comment]
@thorstenhillmann Which issue? There's two in this thread. 

This initial issue:
```
Download failed, retrying: unknown blob
```
when attempting to pull Microsoft/nanoserver on a Linux daemon is to be expected. Linux daemons (including Docker for Windows while in Linux container configuration) do not support pulling Windows images.



The other issue reported in the comments:
```
failed to register layer: re-exec error: exit status 1: output: ProcessBaseLayer
C:\ProgramData\docker\windowsfilter\5f12692222e871fc80f36cf7892b4df0444d78cd211b6c5efd7df19931a131a9: This operation
returned because the timeout period expired.
```
doesn't have enough info for me to help. If this is the issue you are experiencing can you open an issue on https://github.com/docker/docker/issues filling in the template there with all the requested info?

**Comment by thorstenhillmann** [comment]
@darrenstahlmsft question was indeed regarding the second issue.

I´m trying to pull microsoft/nanoserver on a Windows Nano host.

Upfront already some info:

```
[10.0.0.5]:

 PS C:\Windows\System32> docker pull microsoft/nanoserver
Using default tag: latest
latest: Pulling from microsoft/nanoserver
bce2fbc256ea: Pulling fs layer
3ac17e2e6106: Pulling fs layer
bce2fbc256ea: Verifying Checksum
bce2fbc256ea: Download complete
3ac17e2e6106: Verifying Checksum
3ac17e2e6106: Download complete
docker : failed to register layer: re-exec error: exit status 1: output: ProcessBaseLayer 
C:\ProgramData\docker\windowsfilter\6176ffc04b6d44d6fd31c3d2bfb74618d199c386af84a5efe9b98340dcb1ab52: This operation returned because the timeout period expired.
    + CategoryInfo          : NotSpecified: (failed to regis...period expired.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 

[10.0.0.5]: PS C:\Windows\System32> docker version
Client:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64
```


As requested I´ll open an issue

**Comment by dkorinets** [comment]
does the issue "This operation returned because the timeout period expired." was solved?

**Comment by JohanSpannare** [comment]
yes

2017-09-22 13:56 GMT+02:00 Dmitry Korinets <notifications@github.com>:

**Comment by dkorinets** [comment]
Could You tell me how You did it?

**Comment by JohanSpannare** [comment]
You switch to Windows Containers before pulling the image. Windows images
needs Windows Container Mode. And Linux needs Linux Container mode.


2017-09-25 7:46 GMT+02:00 Dmitry Korinets <notifications@github.com>:

**Comment by darrenstahlmsft** [comment]
@dkorinets for the `This operation returned because the timeout period expired` issue, please see https://github.com/moby/moby/issues/27588

I think this issue can be closed, as it has a resolution (use Windows container mode, not Linux container mode), and recent versions of Docker have a better error message to diagnose the problem.

**Comment by dkorinets** [comment]
Kernel Version: 10.0 14393 (14393.1770.amd64fre.rs1_release.170917-1700)
Operating System: Windows 10 Pro
OSType: windows
Architecture: x86_64
CPUs: 4
Total Memory: 7.901GiB

I use Windows container mode, but still have same error 

docker : failed to register layer: re-exec error: exit status 1: output: ProcessBaseLayer C:\ProgramData\Docker\windowsfilter\7b7997d5b4070baf585d700982332cb8b5ee0940ff5dec8b431b46fbad93fc42: This operation returned because the timeout period expired.
At line:1 char:1
+ docker pull microsoft/nanoserver

**Comment by Sawtaytoes** [comment]
Switching to Windows containers in Docker didn't fix the issue for me in Docker `Version 17.09.0-ce-win33 (13620)`.

It doesn't work for `microsoft/nanoserver` or `microsoft/aspnet`.

```shell
PS C:\> docker run -it microsoft/nanoserver
Unable to find image 'microsoft/nanoserver:latest' locally
latest: Pulling from microsoft/nanoserver
bce2fbc256ea: Extracting [==================================================>]  252.7MB/252.7MB
53a0ccfb7e6f: Download complete
C:\Program Files\Docker\Docker\Resources\bin\docker.exe: failed to register layer: re-exec error: exit status 1: output: ProcessBaseLayer C:\ProgramData\Docker\windowsfilter\ad6d37c03cfa6ec711abe544b1fb26492446ccb9276a2e63d5874c7e7704b45d: The parameter is incorrect.
See 'C:\Program Files\Docker\Docker\Resources\bin\docker.exe run --help'.
```

**Comment by bilalmalik777** [comment]
I am using window 10 with 1809 build version, i am unable to pull any image, when i switch to windows container., neither on my local machine nor remote server. But pulling image command work correctly when i switch to linux container. If you need more information let me know
